mdaviz/mainwindow.py

```diff
@@ -76,116 +83,157 @@ def doClose(self, *args, **kw):
 
         self.close()
 
-    # TODO: adapt doOpen to files and folders
     def doOpen(self, *args, **kw):
         """
         User chose to open (connect with) a tiled server.
         """
-        pass
-        # from .tiledserverdialog import TiledServerDialog
-
-        # server_uri = TiledServerDialog.getServer(self)
-        # if not server_uri:
-        #     self.clearContent()
-        # uri_list = self.serverList()
-        # if uri_list[0] == "":
-        #     uri_list[0] = server_uri
-        # else:
-        #     uri_list.insert(0, server_uri)
-        # self.setServers(uri_list)
-
-    def folderName(self):
-        return self._folderName
+        from .opendialog import OpenDialog
+        self.setStatus("Please select a folder...")
+        open_dialog = OpenDialog(self)
```

Comment:
Can this use a standard file dialog from Qt?  Such as [`QtWidgets.QFileDialog.getExistingDirectory()`](https://doc.qt.io/qtforpython-5/PySide2/QtWidgets/QFileDialog.html#PySide2.QtWidgets.PySide2.QtWidgets.QFileDialog.getExistingDirectory)

Reply:
Do you mean that I do not need a UI file for this?

Reply:
You can explain me tomorrow.

Reply:
https://doc.qt.io/qtforpython-5/PySide2/QtCore/Signal.html#PySide2.QtCore.Signal.disconnect